be/src/exec/pipeline/sink/table_function_table_sink_operator.cpp

```diff
@@ -0,0 +1,237 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "table_function_table_sink_operator.h"
+
+#include <utility>
+
+#include "formats/parquet/file_writer.h"
+#include "glog/logging.h"
+
+namespace starrocks::pipeline {
+
+// TODO(letian-jiang): optimize
+StatusOr<std::string> column_to_string(const TypeDescriptor& type_desc, const ColumnPtr& column) {
+    auto datum = column->get(0);
+    if (datum.is_null()) {
+        return "null";
+    }
+
+    switch (type_desc.type) {
+    case TYPE_BOOLEAN: {
+        return datum.get_uint8() ? "true" : "false";
+    }
+    case TYPE_TINYINT: {
+        return std::to_string(datum.get_int8());
+    }
+    case TYPE_SMALLINT: {
+        return std::to_string(datum.get_int16());
+    }
+    case TYPE_INT: {
+        return std::to_string(datum.get_int32());
+    }
+    case TYPE_BIGINT: {
+        return std::to_string(datum.get_int64());
+    }
+    case TYPE_DATE: {
+        return datum.get_date().to_string();
+    }
+    case TYPE_DATETIME: {
+        return datum.get_timestamp().to_string();
+    }
+    case TYPE_CHAR:
+    case TYPE_VARCHAR: {
+        return datum.get_slice().to_string();
+    }
+    default: {
+        return Status::InvalidArgument("unsupported partition column type" + type_desc.debug_string());
+    }
+    }
+}
+
+inline std::string get_partition_location(const string& path, const std::vector<std::string>& names,
+                                          const std::vector<std::string>& values) {
+    std::string partition_location = path;
+    for (size_t i = 0; i < names.size(); i++) {
+        partition_location += names[i] + "=" + values[i] + "/";
```

Comment:
what if string value contains `"/"` and `=`

Reply:
good point

Reply:
Update: use URL encoding